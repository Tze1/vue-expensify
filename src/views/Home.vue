<template>
  <div v-if="userLoggedIn" class="home component">
    <h2>Home</h2>
    <p>This is the Home view.</p>
    <HelloWorld :msg="`Hello, ${user.displayName.split(' ')[0]}`"/>
  </div>
</template>

<script>
  import { mapState } from 'vuex';
  import HelloWorld from '../components/HelloWorld';

  export default {
    name: 'Home',
    computed: mapState({
      userLoggedIn () {
        return this.$store.getters.userLoggedIn;
      },
      user: state => state.auth.user,
    }),
    components: {
      HelloWorld
    },
  };
</script>

<style scoped lang="scss">
  @import '../styles/app.scss';
</style>
