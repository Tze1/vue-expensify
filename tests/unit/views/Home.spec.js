import { shallowMount } from '@vue/test-utils'
import Home from '@/views/Home'

describe('Home.vue', () => {
  it('renders Home view properly', () => {
    const $store = {
      state: {
        auth: {
          user: {
            displayName: 'Test',
            photoUrl: '/img/avatar-default.png',
          },
        },
      },
      getters: {
        userLoggedIn: state => {
          return (state.user !== undefined);
        },
      },
    };
    const wrapper = shallowMount(Home, {
      mocks: {
        $store,
      },
    });

    expect(wrapper).toMatchSnapshot();
  })
})
