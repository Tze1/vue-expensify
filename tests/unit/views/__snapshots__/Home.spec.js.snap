// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue renders Home view properly 1`] = `
<div class="home component">
  <h2>Home</h2>
  <p>This is the Home view.</p>
  <helloworld-stub msg="Hello, Test"></helloworld-stub>
</div>
`;
